lib/src/main/java/me/onebone/toolbar/ScrollStrategy.kt

```diff
@@ -246,6 +282,32 @@ internal class ExitUntilCollapsedNestedScrollConnection(
 			velocity
 		}
 
+		// TODO: Cancel expand/collapse animation inside onPreScroll
+		snapConfig?.let { scaffoldState.toolbarState.performSnap(it) }
+
 		return available.copy(y = available.y - left)
 	}
 }
+
+// TODO: Is there a better solution rather OptIn ExperimentalToolbarApi?
+@OptIn(ExperimentalToolbarApi::class)
+private suspend fun CollapsingToolbarState.performSnap(snapConfig: SnapConfig) {
+	if (progress > snapConfig.edge && progress < 1f) {
+		expand(snapConfig.expandDuration)
+	} else if (progress <= snapConfig.edge && progress > 0f){
+		collapse(snapConfig.collapseDuration)
+	}
+}
+
+// TODO: Is there a better solution rather OptIn ExperimentalToolbarApi?
+@OptIn(ExperimentalToolbarApi::class)
```

Comment:
`expandOffset` and `collapseOffset` are experimental because it is subject to change their behavior or surface anytime. Meantime, `performSnap` is private and we don't have to be warned by this potential change so I think it would be sufficient.